src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1076,6 +1080,16 @@ def delete_policy(cmd, client, resource_group_name, vault_name, object_id=None,
                                     tags=vault.tags,
                                     properties=vault.properties),
                                 no_wait=no_wait)
+
+
+def _check_application_id_match(application_id, existing_application_id):
+    if application_id and not existing_application_id:
+        return False
+    if not application_id and existing_application_id:
+        return False
+    if not application_id and not existing_application_id:
+        return True
+    return application_id.lower() == existing_application_id.lower()
```

Comment:
`(application_id or '').lower() == (p.application_id or '').lower()` should do it.

Reply:
done